feat(nextjs): Add `excludeServerRoutes` option to manual setup page (#5789)
getsentry/sentry-docs: src/platforms/javascript/guides/nextjs/manual-setup.mdx

```

(This assumes you are using [the `next.config.js` setup shown above](#extend-nextjs-configuration).)

> ## Configure Serverside Auto-instrumentation

Reviewer: ```suggestion
## Configure Server-side Auto-instrumentation
```

The SDK will automatically instrument API routes and server-side [Next.js data fetching methods](https://nextjs.org/docs/basic-features/data-fetching/overview) with error and performance monitoring.

### Disable API Route and Data Fetching Auto-instrumentation Entirely

_(New in version 7.14.0)_

> To disable the automatic instrumentation of API route handlers and serverside data fetching functions, set the `autoInstrumentServerFunctions` to `false`.

Reviewer: ```suggestion
To disable the automatic instrumentation of API route handlers and server-side data fetching functions, set the `autoInstrumentServerFunctions` to `false`.
```

```javascript {filename:next.config.js}
const moduleExports = {
  sentry: {
    autoInstrumentServerFunctions: false,
  },
};
```

> Under the hood, when using this option, the SDK is using a Webpack loader to wrap all your API route handlers and data fetching methods.

Reviewer: ```suggestion
With this option, under the hood, the SDK is using a Webpack loader to wrap all your API route handlers and data fetching methods.
```

### Opt In to Auto-instrumentation on Specific Routes

_(New in version 7.14.0)_

> If the automatic instrumentation doesn't work for your use case, you can turn it off globally and choose instead to only wrap specific API route handlers or data fetching functions.

Reviewer: ```suggestion
If the automatic instrumentation doesn't work for your use case, you can turn it off globally and choose to only wrap specific API route handlers or data fetching functions instead.
```

const handler = (req, res) => {
  res.status(200).json({ name: "John Doe" });
};

export default withSentry(handler);
```

```typescript {filename:pages/api/*}
import type { NextApiRequest, NextApiResponse } from "next"
import { withSentry } from "@sentry/nextjs";

const handler = (req: NextApiRequest, res: NextApiResponse) => {
  res.status(200).json({ name: "John Doe" });
};

export default withSentry(handler);
```

For data fetching methods, use the following functions:

- `withSentryServerSideGetInitialProps` for `getInitialProps`
- `withSentryGetServerSideProps` for `getServerSideProps`
- `withSentryGetStaticProps` for `getStaticProps`
- `withSentryServerSideErrorGetInitialProps` for `getInitialProps` in [custom Error pages](https://nextjs.org/docs/advanced-features/custom-error-page)
- `withSentryServerSideAppGetInitialProps` for `getInitialProps` in [custom `App` components](https://nextjs.org/docs/advanced-features/custom-app)
- `withSentryServerSideDocumentGetInitialProps` for `getInitialProps` in [custom `Document` components](https://nextjs.org/docs/advanced-features/custom-document)

### Opt Out of Auto-instrumentation on Specific Routes

_(New in version 7.20.0)_

> If you want auto-instrumenatation to apply by default, but want to exclude certain routes, use the `excludeServerRoutes` option in the `sentry` object in your `next.config.js`:
> 

Reviewer: ```suggestion
If you want auto-instrumenatation to apply by default, but also want to exclude certain routes, use the `excludeServerRoutes` option in the `sentry` object in your `next.config.js`:

```

```javascript {filename:next.config.js}
const moduleExports = {
  sentry: {
    excludeServerRoutes: [
      "/some/excluded/route",
      "/excluded/route/with/[parameter]",
      /^\/route\/beginning\/with\/some\/prefix/,
      /\/routeContainingASpecificPathSegment\/?/,
    ],
  },
};
```

> Excluded routes can be specified either as strings (which must exactly match the full route you want to exclude, with a leading slash but no trailing one) or regexes.

Reviewer: ```suggestion
Excluded routes can be specified either as regexes or strings, (which must match the full route you want to exclude exactly and have a leading slash, but no trailing one).
```

Author: ```suggestion
Excluded routes can be specified either as regexes or strings. When using a string, make sure that it matches the route exactly, and has a leading slash but no trailing one.
```
